packages/client/src/views/Settings/items/Language.tsx

```diff
@@ -0,0 +1,172 @@
+ * graphic logo are (registered/a) trademark(s) of Plan International.
+ */
+import * as React from 'react'
+import {
+  ListViewItemSimplified,
+  ResponsiveModal,
+  FloatingNotification,
+  NOTIFICATION_TYPE
+} from '@opencrvs/components/lib/interface'
+import { useIntl, FormattedMessage } from 'react-intl'
+import {
+  userMessages,
+  buttonMessages,
+  constantsMessages
+} from '@client/i18n/messages'
+import {
+  LabelContainer,
+  ValueContainer,
+  DynamicHeightLinkButton,
+  Message,
+  Label
+} from '@client/views/Settings/items/utils'
+import { useSelector, useDispatch } from 'react-redux'
+import { IStoreState } from '@client/store'
+import { ILanguageState } from '@client/i18n/reducer'
+import { TertiaryButton, PrimaryButton } from '@opencrvs/components/lib/buttons'
+import { Select } from '@opencrvs/components/lib/forms'
+import { getAvailableLanguages } from '@client/i18n/utils'
+import { IUserDetails } from '@client/utils/userUtils'
+import { getUserDetails } from '@client/profile/profileSelectors'
+import { changeLanguage as changeLanguageActionCreator } from '@client/i18n/actions'
+
+interface ILanguageOptions {
+  [key: string]: string
+}
+
+export function Language() {
+  const [showLanguageSettings, setShowLanguageSettings] =
+    React.useState<boolean>(false)
+  const [showSuccessNotification, setShowSuccessNotification] =
+    React.useState<boolean>(false)
+  const intl = useIntl()
+  const languages = useSelector<IStoreState, ILanguageState>(
+    (state) => state.i18n.languages
+  )
+  const language = useSelector<IStoreState, string>(
+    (state) => state.i18n.language
+  )
+  const [selectedLanguage, setSelectedLanguage] =
+    React.useState<string>(language)
+  const toggleLanguageSettingsModal = React.useCallback(() => {
+    setShowLanguageSettings((prevValue) => !prevValue)
+  }, [])
+
+  const cancelLanguageSettings = React.useCallback(() => {
+    setSelectedLanguage(language)
+    setShowLanguageSettings((prevValue) => !prevValue)
+  }, [language])
+
+  const toggleSuccessNotification = React.useCallback(() => {
+    setShowSuccessNotification((prevValue) => !prevValue)
+  }, [])
+
+  const userDetails = useSelector<IStoreState, IUserDetails | null>(
+    getUserDetails
+  )
+
+  const dispatch = useDispatch()
+
+  const changeLanguage = React.useCallback(() => {
+    if (userDetails) {
+      dispatch(changeLanguageActionCreator({ language: selectedLanguage }))
+      toggleLanguageSettingsModal()
+      toggleSuccessNotification()
+    }
+  }, [
+    userDetails,
+    dispatch,
+    toggleLanguageSettingsModal,
+    toggleSuccessNotification,
+    selectedLanguage
+  ])
+
+  const langChoice = [] as ILanguageOptions[]
+  const availableLangs = getAvailableLanguages()
+  availableLangs.forEach((lang: string) => {
+    if (languages[lang]) {
+      langChoice.push({
+        value: lang,
+        label: languages[lang].displayName
+      })
+    }
+  })
```

Comment:
We can probably extract this into a custom hook that returns the availableLanguages and displayName of the currently selected language

Reply:
The refactor as discussed is done in https://github.com/opencrvs/opencrvs-core/pull/3647/commits/4d5fcbd4d4a3809250db0d3e9e9c2b3d8741a424